pkg/gui/controllers/commits_files_controller.go

```diff
@@ -176,43 +176,56 @@ func (self *CommitFilesController) checkout(node *filetree.CommitFileNode) error
 	return self.c.Refresh(types.RefreshOptions{Mode: types.ASYNC})
 }
 
-func (self *CommitFilesController) discard(node *filetree.CommitFileNode) error {
+func (self *CommitFilesController) discard(selectedNodes []*filetree.CommitFileNode) error {
 	parentContext, ok := self.c.CurrentContext().GetParentContext()
 	if !ok || parentContext.GetKey() != context.LOCAL_COMMITS_CONTEXT_KEY {
 		return self.c.ErrorMsg(self.c.Tr.CanOnlyDiscardFromLocalCommits)
 	}
 
-	if node.File == nil {
-		return self.c.ErrorMsg(self.c.Tr.DiscardNotSupportedForDirectory)
-	}
-
 	if ok, err := self.c.Helpers().PatchBuilding.ValidateNormalWorkingTreeState(); !ok {
 		return err
 	}
 
-	prompt := self.c.Tr.DiscardFileChangesPrompt
-	if node.File.Added() {
-		prompt = self.c.Tr.DiscardAddedFileChangesPrompt
-	} else if node.File.Deleted() {
-		prompt = self.c.Tr.DiscardDeletedFileChangesPrompt
-	}
+	removeFileRange := func() error {
+		return self.c.WithWaitingStatus(self.c.Tr.RebasingStatus, func(gocui.Task) error {
+			selectedNodes = normalisedSelectedCommitFileNodes(selectedNodes)
+
+			return self.c.Confirm(types.ConfirmOpts{
+				Title:  self.c.Tr.DiscardFileChangesTitle,
+				Prompt: self.c.Tr.DiscardFileChangesPrompt,
+				HandleConfirm: func() error {
+					var filePaths []string
```

Comment:
I need to build a list of all the individual files that are covered under `selectedNodes`. I'm feeling like there might be a better way to do this, but I didn't see a clean way to do it without just replicating the recursion that's already in `node.ForEachFile`.